pkg/apis/garden/validation/validation.go

```diff
@@ -39,6 +39,28 @@ import (
 	"k8s.io/apimachinery/pkg/util/validation/field"
 )
 
+type dnsSet map[garden.DNSProvider]struct{}
+
+func (s dnsSet) list() []string {
+	r := []string{}
+	for k := range s {
+		r = append(r, string(k))
+	}
+	return r
+}
+
+var availableDNS dnsSet
```

Comment:
Why not using Sets (`k8s.io/apimachinery/pkg/util/sets`), already providing `Insert`, `Has` and `List` methods?